\documentclass[a4paper,10pt]{article}

% packages
\usepackage[utf8]{inputenc} % allows usage of spanish special characters
\usepackage[spanish,english]{babel} % english dictionary for proper hyphenation
\usepackage{amsmath} % math expressions
\usepackage{upgreek} % upright greek letters for math
\usepackage{txfonts} % nice fonts 
\usepackage{authblk} % author affiliations
\usepackage{graphicx} % images
\usepackage{float} % image positioning
\usepackage{multirow} % allows merging cells in tables
\usepackage{makecell} % more table customization
\usepackage{mathtools} % nice matrices
\usepackage{units} % nice fractions
\usepackage{rotating} % rotated text
\usepackage{caption} % customization of figure/table captions
\usepackage{subcaption} % collages of multiple images
\usepackage{hyperref} % hyperlinks
\usepackage{nameref} % cross-references between draft and supplementary material
\usepackage{zref-xr,zref-user} % more cross-references formatting
\zxrsetup{toltxlabel}
\zexternaldocument*{supp}
\usepackage[square,numbers,sort&compress]{natbib}
\usepackage[table,xcdraw]{xcolor} % text and table background colors
\usepackage[margin=3cm]{geometry} % margins
\usepackage{lineno} % line numbers
\usepackage{textcomp} % special characters (some references break if this isn't loaded?)

\usepackage{pdflscape} %% Landscape
\usepackage{dcolumn}
\usepackage{booktabs}
\usepackage{savesym}
\savesymbol{iint}
\restoresymbol{TXF}{iint}
%% \usepackage{pdfpages} %% to include a pdf file

% customization
\setcounter{Maxaffil}{0} % affiliations
\renewcommand\Affilfont{\itshape\small} % style of affiliations text
\makeatletter\renewcommand{\@biblabel}[1]{#1.}\makeatother % change [number] for number. in reference list
\addto{\captionsenglish}{\renewcommand{\bibname}{References}}
\newcommand{\idea}[1]{\textcolor{red}{#1}}
\newcommand{\mr}{\textcolor{red}{\textbf{[missing ref(s)]}}}
\newcommand{\Burl}[1]{\url{#1}}
\zxrsetup{toltxlabel} % cross references: draft & suppl mat
\zexternaldocument*[M-]{draft}
\hypersetup{
  colorlinks = true,
  citecolor=  black,
  linkcolor = {blue},
  filecolor = cyan % controls color of external ref, if used
}
\captionsetup[figure]{font=small, labelfont={bf}, labelformat={default},
   labelsep=period, name={Figure}} % custom figure captions
\captionsetup[table]{font=small, labelfont={bf}, labelformat={default},
   labelsep=period, name={Table}} % custom table captions
\hypersetup{citecolor={blue}}
\definecolor{lightgray}{gray}{0.9}
\newcommand{\figref}[2][]{%
  \hyperref[{#2}]{%
    Figure~\ref*{#2}%
    \ifx\\#1\\%
    \else
      #1%
    \fi
  }%
}
\newcommand{\tableref}[1]{%
  \hyperref[{#1}]{%
   Table~\ref*{#1}%
  }%
}
\newcommand{\methodsref}[1]{%
  \hyperref[{methods:#1}]{%
   Methods:~\nameref*{methods:#1}%
  }%
}

% title, authors, affiliations
\title{Provisional Title}
\author[1,2,3]{Authors}
\author[1,2,$\dagger$]{Álvaro Sánchez}
\affil[1]{Department of Ecology \& Evolutionary Biology,
Yale University, New Haven, CT, USA}
\affil[2]{Microbial Sciences Institute,
Yale University, New Haven, CT, USA}
\affil[3]{Other affiliations...}
\affil[$\dagger$]{To whom correspondence should be addressed: \normalfont alvaro.sanchez@yale.edu}
\date{}



  
\begin{document}

%\linenumbers

\maketitle

\begin{abstract}
  
The abstract goes here.
  
\end{abstract}

\section*{Introduction}\label{intro}

Microbial communities often invade one another.
This has been observed, for instance, 
in river courses where terrestrial microbial communities mix with aquatic microorganisms
\cite{Mansour2018,Luo2020,Vass2021}
or in soil communities being invaded as a result of
tillage and outplanting \cite{Rillig2016} or by
aerially dispersed bacteria and funghi \cite{Evans2019}.
The human digestive system can get invaded several times a day by the microbial consortia
that reside on the ingested food, and the skin microbiota is also subject to
invasions when making contact with environmental sources of microbes \cite{Vandegrift2019}.

The phenomenon by which entire microbiomes invade one another has been termed
\textit{community coalescence} \cite{Rillig2015}.
Ecologists have long contemplated the idea that interactions between multiple co-invading species
can produce correlated invasional outcomes
\cite{Gilpin1994,Grosholz2005,Green2011,Livingston2013,Rillig2015,OLoughlin2017,Castledine2020}.
One example is the hypothesis known as \textit{invasional meltdown}, which proposes that
positive interactions between co-invading species can enhance their invasive success and
facilitate future invasions \cite{Simberloff1999,Simberloff2006,Gurevitch2006}.
However, and
in spite of its clear potential importance, the role of coalescence in microbiome assembly is
only beginning to be addressed and little is known about the mechanisms that govern it and its
potential implications.
Early mathematical models of community-community invasions \cite{Gilpin1994,Toquenaga1997}
as well as more recent work
\cite{Tikhonov2016,Tikhonov2017,Vila2019,Lechon2021}
suggest that high-order invasion effects are common
during community coalescence. Communities that have a previous history of coexistence may exhibit an
emergent ``cohesiveness'' which produces correlated invasional outcomes among species from the
same community \cite{Livingston2013,Sierocinski2017}.
The situation where ecological partners in the invading community recruit each other into the final
coalesced community has been called \textit{ecological co-selection} \cite{Rillig2017,Sierocinski2017}.

The mechanisms of ecological co-selection during community coalescence are still poorly understood.
Do a few key species recruit everyone else, or are collective interactions among all species
(including the rarer members of the community) relevant for coalescence outcomes?
While it is reasonable to expect species with larger population sizes to have a proportionally oversized effect,
natural communities tend to be highly diverse \cite{Louca2016}
and
the role played by the less abundant species has long been subject to debate \cite{Winfree2015}.
Laboratory cultures have also been found to contain uneven distributions of multiple strains that feed
off the metabolic secretions of the dominant species \cite{Rosenzweig1994,Goldford2018}.
The fate of these sub-dominant taxa may be dependent on the invasional success of their dominant
species, or, alternatively, the dominant itself may owe its dominance (at least in part) to cross-feeding
or other forms of facilitation from the rarer members of the population.
These scenarios would give rise to ``top-down'' or ``bottom-up'' community cohesiveness, respectively.
Either of these forms of co-selection could, in principle, be positive (recruitment) or negative (antagonism),
as illustrated in \figref[e]{fig1}.
Which of these situations are typically found in nature?
Addressing this question has been experimentally challenging in the past \cite{Rillig2017,Sierocinski2017}.

In previous work, we have shown that a large amount of soil and plant microbiomes can be cultured
\textit{ex situ} in synthetic minimal environments with a single supplied limiting resource under serial
growth-dilution cycles \cite{Goldford2018} (\figref[a-b]{fig1}). Under these conditions, environmental
microbiomes spontaneously re-assemble into complex multi-species communities sustained by dense
cross-feeding facilitation networks \cite{Goldford2018}. In addition,
and just like in natural assemblies,
species abundance distributions
in these communities are generally long-tailed and uneven (\figref[d]{fig1} and \figref{figS1}),
with the dominant (most abundant) species typically comprising most of the biomass ($\text{median}=46\%$,
\figref{figS1}). Because these communities are easy to manipulate and grow in high throughput, and
are largely made up by culturable members, they represent good test cases to investigate ecological
co-selection during community coalescence. Here we focus on the dominants and ask whether they can
co-select or be co-selected by the sub-dominant species in their communities (henceforth referred to as
their \textit{cohorts}, \figref[c]{fig1}).
Our results confirm that positive top-down co-selection is common, but its effects are weak.
In contrast, bottom-up co-selection can be very strong, and positive co-selection is far more common
than negative co-selection.
We then turn to a microbial consumer-resource model (microCRM)
\cite{Goldford2018,Marsland2019,Marsland2020}
that is able to capture the dynamics of microbial communities dominated by metabolic interactions,
as is the case for the ones assembled in our experimental conditions \cite{Goldford2018}.
We show that the empirically observed trends in ecological co-selection are reproduced
with minimal model assumptions, and that
tuning the complexity of the metabolic interactions in our \textit{in silico} communities can modulate
the recurrence of top-down or bottom-up co-selection.
Our findings indicate that collective interactions play an important role at dictating community structure
during coalescence.

\section*{Results \& Discussion}\label{results-discussion}

We collected eight natural microbiomes from different soil and plant environmental samples
(\figref[a]{fig1})
and used them to inoculate our synthetic communities, which were stabilized 
in serial batch-culture biorreactors for 84 generations in synthetic minimal media containing either
glutamine or citrate as the only supplied carbon source
(\figref[b]{fig1}, \methodsref{community-assembly}).
We isolated the dominant species of every community
(\methodsref{dominants})
and identified them by Sanger-sequencing their 16S rRNA gene
(\methodsref{sequencing}), which correctly matched the dominant
Exact Sequence Variant (ESV) \cite{Callahan2016,Callahan2017} found through
community-level 16S illumina sequencing
(\figref{figS1}).
These dominants remained at high frequency after seven additional transfers
with the exception of two of the citrate communities and one of the glutamine
communities (where the dominants were presumably a transiently dominating species)
that were excluded from further analysis
(\figref{figS1}).
Similarly, pairs of communities where the dominants shared a same 16S sequence
and had similar colony morphology were excluded
(\figref{figS1}).

\subsubsection*{The strength of top-down co-selection is dependent on the primary supplied resource}

If communities being coalesced were highly cohesive from the top-down,
the dominant species would co-select the rarer members of its community during coalescence
(\figref[e]{fig1}, left panels).
In this scenario, we would expect the outcome of community coalescence to be predicted by
which of the two dominants is most competitive in pairwise competition.
Coalescence outcomes can be quantified by the similarity between the coalesced and the
invasive communities
(\methodsref{metrics}).
To test this hypothesis, we performed all pairwise competitions between dominant species
in glutamine and citrate environments by mixing them 1:1 on their native media and propagating
the cultures for seven serial transfers, roughly 42 generations
(\methodsref{competitions}).
We then performed all possible pairwise community coalescence experiments by mixing
equal volumes of the communities and propagating the resulting cultures for seven extra
transfers
(\figref[f]{fig1}).
The frequencies of all species in both community-community and dominant-dominant competitions
were determined by 16S illumina sequencing
(\methodsref{sequencing}).

We found that, for communities assembled in the glutamine environment,
the pairwise competitive ability of an invasive dominant
is only weakly predictive of the performance of the invasive community in coalescence
(as quantified by the relative Bray-Curtis similarity between the coalesced and invasive communities),
but this correlation was stronger for the citrate communities
(\figref[a]{fig2}, $R^2=0.15$, $p<0.05$ for glutamine and $R^2=0.57$, $p<10^{-4}$ for citrate).
Alternative quantifications of community distance yield similar results,
with weaker effects when the metric used
accounts only for the presence/absence of specific species and not for their relative abundance in
the communities
(\figref{figS2}).
All these metrics include the presence of the dominant species themselves. To better disentangle the effect
that these dominants have on the other member of their communities, we repeated the analysis this time
excluding the dominant species from the compositional data, finding that our results still hold
(\figref{figS3}).
Furthermore, we observed that, in the glutamine communities,
the relative frequency of a dominant against another in head-to-head pairwise competition
is barely predictive of its relative frequency against that same other dominant when the
cohorts are present too, i.e. after coalescence
($R^2=0.04$, $p>0.05$, \figref{figS4}).
On the other hand, pairwise competition between dominants of the citrate communities
is highly correlated with the relative abundance of said dominants in community coalescence
 ($R^2=0.83$, $p<10^{-8}$, \figref{figS4}).

Together, these results suggest that the strength of top-down co-selection depends on
the primary resource supplied to the coalesced communities.
Communities assembled with citrate as the primary supplied resource
display a strong degree of top-down co-selection,
with the fates of the sub-dominant species determined to a large extent by dominant-dominant
pairwise competition. This competition is, in turn, only weakly affected by the presence of the cohorts.
For glutamine communities, although some degree of top-down co-selection is consistent with our data,
the cohorts do not appear to be passively responding to their dominants.
Instead, evidence suggests that the cohorts might be playing an active role in community coalescence.
This finding led us to investigate the potential role of bottom-up ecological co-selection
(\figref[e]{fig1}, right panels), i.e. whether in some cases the dominants may be co-selected
for or against by their cohorts.

\clearpage

\section*{Methods}\label{methods}

\subsubsection*{Stabilization of environmental communities in simple synthetic environments}
\label{methods:community-assembly}

Communities were stabilized \textit{ex situ} as described in \cite{Goldford2018}.
In short, environmental samples (soil, leaves...) within one meter radius in eight different
geographical locations were collected with sterile
tweezers or spatulas into 50mL sterile tubes (\figref[a]{fig1}).
One gram of each sample was allowed to
sit at room temperature in 10mL of phosphate buffered saline (1$\times$PBS) containing
200$\upmu$g/mL cycloheximide to suppress eukaryotic growth.
After 48h, samples were mixed 1:1 with 80\% glycerol and kept frozen at $-80^\circ$C.
Starting microbial communities were prepared by scrapping the frozen stocks into
$200\upmu$L of 1$\times$PBS and adding a volume of $4\upmu$L to $500\upmu$L
of synthetic minimal media (1$\times$M9) supplemented with $200\upmu$g/mL cycloheximide
and 0.07 C-mol/L glutamine or sodium citrate as the carbon source in 96 deep-well plates
(1.2mL; VWR).
Cultures were then incubated still at $30^\circ$C to allow for re-growth.
After 48h, samples were fully homogenized and biomass increase was followed by measuring
the optical density (620nm) of $100\upmu$L of the cultures in a Multiskan FC plate reader
(Thermo Scientific).
Communities were stabilized \cite{Goldford2018} by passaging $4\upmu$L of the cultures into
$500\upmu$L of fresh media (1$\times$M9 with the carbon source)  every 48h for a total of
12 transfers at a dilution factor of 1:100,
roughly equivalent to 80 generations per culture (\figref[b]{fig1}).
Cycloheximide was not added to the media after the first two transfers.

\subsubsection*{Isolation of dominant species}\label{methods:dominants}

For each community, the most abundant colony morphotype at the end of the ninth transfer
was selected, resuspended in $100\upmu$L 1$\times$PBS and serially diluted (1:10).
Next, $20\upmu$L of the cells diluted to $10^{-6}$ were plated in the corresponding synthetic
minimal media and allowed to regrow at $30^\circ$C for 48h. Dominants were then identified
(\figref[c]{fig1}),
inoculated into $500\upmu$L of fresh media and incubated still at $30^\circ$C for 48h.
After this period, the communities stabilized for eleven transfers and the isolated dominants
were ready for the competition experiments at the onset of the twelfth transfer.

\subsubsection*{Dominant-dominant and community-community competitions}
\label{methods:competitions}

All possible pairwise dominant-dominant and community-community
competition experiments
were performed by mixing equal volumes ($4\upmu$L) of each of the eight
communities or eight dominants at the onset of the twelfth transfer.
Competitions were set up in their native media,
i.e. in $500\upmu$L of 1$\times$M9 supplemented with 0.07 C-mol/L of
either glutamine or citrate in 96 deep-well plates.
Plates were incubated at $30^\circ$C for 48h.
Pairwise competitions were further propagated for seven serial transfers
(roughly 42 generations, \figref[f]{fig1}) by transferring $8\upmu$L of
each culture to fresh media ($500\upmu$L).

\subsubsection*{Determination of community composition by 16S sequencing}
\label{methods:sequencing}

The sequencing protocol was identical to that described in \cite{Goldford2018}.
Community samples were collected by spinning down at 3500rpm for 25min
in a bench-top centrifuge at room temperature;
cell pellets were stored at $-80^\circ$C before processing.
To maximize Gram-positive bacteria cell wall lysis,
the cell pellets were re-suspended and incubated at $37^\circ$C for 30min
in enzymatic lysis buffer (20mM Tris-HCl, 2mM sodium EDTA, 1.2\% Triton X-100)
and 20mg/mL of lysozyme from chicken egg white (Sigma-Aldrich).
After cell lysis, the DNA extraction and purification was performed using the
DNeasy 96 protocol for animal tissues (Qiagen).
The clean DNA in $100\upmu$L elution buffer of 10mM Tris-HCl, 0.5mM EDTA
at pH 9.0 was quantified using Quan-iT PicoGreen dsDNA Assay Kit
(Molecular Probes, Inc.)
and normalized to 5ng/$\upmu$L in nuclease-free water (Qiagen)
for subsequent 16S rRNA illumina sequencing.
16S rRNA amplicon library preparation was performed following a dual-index
paired-end approach \cite{Kozich2013}.
Briefly, PCR amplicon libraries of V4 regions of the 16S rRNA were prepared 
sing dual-index primers (F515/R805), then pooled and sequenced
using the Illumina MiSeq chemistry and platform.
Each sample went through a 30-cycle PCR in duplicate of $20\upmu$L
reaction volumes using 5ng of DNA each, dual index primers, and AccuPrime Pfx SuperMix (Invitrogen).
The thermocycling procedure includes a 2min initial denaturation step at
$95^\circ$C, and 30 cycles of the following PCR scheme:
(a) 20-second denaturation at $95^\circ$C,
(b) 15-second annealing at $55^\circ$C,
and (c) 5-minute extension at $72^\circ$C.
The duplicate PCR products of each sample were pooled, purified, and normalized
using SequalPrep PCR cleanup and normalization kit (Invitrogen).
Barcoded amplicon libraries were then pooled and sequenced using
Illumina Miseq v2 reagent kit, which generated 2$\times$250bp paired-end reads
at the Yale Center for Genome Analysis (YCGA).
The sequencing reads were demultiplexed on QIIME 1.9.0 \cite{Caporaso2010}.
The barcodes, indexes, and primers were removed from raw reads,
producing FASTQ files with both the forward and reverse reads for each sample,
ready for DADA2 analysis \cite{Callahan2017}.
DADA2 version 1.1.6 was used to infer unique biological exact sequence variants
(ESVs) for each sample
and na{\"i}ve Bayes was used to assign taxonomy using the SILVA version 123
database \cite{Wang2007,Quast2013}.

\subsubsection*{Metrics of community distance}
\label{methods:metrics}

Beta-diversity indexes between the invasive and coalesced communities
or the resident and coalesced communities were computed using various
similarity metrics. For two arbitrary communities with
ESV abundances represented by the vectors
$\mathbf{x} = \left( x_1, x_2, \cdots, x_N \right)$
and
$\mathbf{y} = \left( y_1, y_2, \cdots, y_N \right)$
(where $x_i$ and $y_i$ represent the relative abundance of the $i$th ESV in
each community respectively and $N$ is the total number of ESVs), the Bray-Curtis
similarity $BC \left( \mathbf{x}, \mathbf{y} \right)$ is calculated as
\cite{Bray1957}

\begin{equation}
BC \left( \mathbf{x}, \mathbf{y} \right) = 
\sum_i \mathrm{min} \left( x_i , y_i \right)
\label{eq:bray-curtis}
\end{equation}
%
The Jensen-Shannon similarity
$JS \left( \mathbf{x}, \mathbf{y} \right)$
is defined as one minus the Jensen-Shannon distance (which is, in turn,
the square root of the Jensen-Shannon divergence \cite{Lin1991})

\begin{equation}
JS \left( \mathbf{x}, \mathbf{y} \right) = 
1 - \sqrt{\frac{1}{2}KL \left( \mathbf{x}, \mathbf{m} \right) +
\frac{1}{2}KL \left( \mathbf{y}, \mathbf{m} \right)}
\label{eq:jensen-shannon}
\end{equation}
%
where $\mathbf{m} = \left( \mathbf{x} + \mathbf{y} \right)/2$ and KL denotes the
Kullback-Leibler divergence \cite{Kullback1951}

\begin{equation}
KL \left( \mathbf{x}, \mathbf{y} \right) = 
\sum_i x_i \; \mathrm{log_2} \left( \frac{x_i}{y_i} \right)
\label{eq:kullback-leibler}
\end{equation}
%
The Jaccard similarity is given by
$J \left( \mathbf{x}, \mathbf{y} \right)$ \cite{Jaccard1912}

\begin{equation}
J \left( \mathbf{x}, \mathbf{y} \right) = 
\frac{\left|\; \mathbf{x} \cap \mathbf{y} \;\right|}
{\left|\; \mathbf{x} \cup \mathbf{y} \;\right|}
\label{eq:jaccard}
\end{equation}
%
Additionally, we quantify coalescence outcomes by examining the fraction of the endemic
cohort of the original communities that persists in the coalesced one. We call
$E \left( \mathbf{x}, \mathbf{y} \right)$ to the fraction of endemic species of
$\mathbf{x}$ that are also found in $\mathbf{y}$.

\iffalse

We denote this
metric as
$E \left( \mathbf{x}_\mathrm{I},
\mathbf{x}_\mathrm{R},
\mathbf{x}_\mathrm{C} \right)$.

\begin{equation}
E \left( \mathbf{x}_\mathrm{I},
\mathbf{x}_\mathrm{R},
\mathbf{x}_\mathrm{C} \right) = 
\left.
\sum_i
f \left( x_{\mathrm{I}i}, x_{\mathrm{R}i}, x_{\mathrm{C}i} \right)
\;
\middle/
\;
\sum_i
g \left( x_{\mathrm{I}i}, x_{\mathrm{R}i} \right)
\right.
\label{eq:endemic}
\end{equation}
%
where we have defined

\begin{equation}
\begin{split}
f \left( x_{\mathrm{I}i}, x_{\mathrm{R}i}, x_{\mathrm{C}i} \right) & =
\begin{cases}
    1 & \text{if } x_{\mathrm{I}i} > 0 \text{ and }
                   x_{\mathrm{R}i} = 0 \text{ and }
                   x_{\mathrm{C}i} > 0 \\
    0 & \text{otherwise}
\end{cases} \\
g \left( x_{\mathrm{I}i}, x_{\mathrm{R}i} \right) & =
\begin{cases}
    1 & \text{if } x_{\mathrm{I}i} > 0 \text{ and }
                   x_{\mathrm{R}i} = 0 \\
    0 & \text{otherwise}
\end{cases}
\end{split}
\label{eq:endemic-f}
\end{equation}

\fi

For all the metrics above, we quantify the relative similarity between the invasive
and the coalesced communities using relative metrics ($Q$):

\begin{equation}
Q \left( \mathbf{x}_\mathrm{I},
\mathbf{x}_\mathrm{R},
\mathbf{x}_\mathrm{C} \right) = 
\frac{F \left( \mathbf{x}_\mathrm{I},\mathbf{x}_\mathrm{C}\right)}
{F \left( \mathbf{x}_\mathrm{I},\mathbf{x}_\mathrm{C}\right)
+
F \left( \mathbf{x}_\mathrm{R},\mathbf{x}_\mathrm{C}\right)}
\label{eq:q}
\end{equation}
%
where the subindices I, R and C correspond to the invasive, resident and coalesced
communities respectively, and $F$ represents one of $BC$ (Bray-Curtis similarity), $JS$
(Jensen-Shannon similarity), $J$ (Jaccard similarity) or $E$ (endemic survival) defined above.

\subsubsection*{Simulations}\label{methods:sim}

We used the Community Simulator package \cite{Marsland2020} and included new
features for our simulations. In the package,
species are characterized by their resource uptake rates ($c_{i\alpha}$ for
species $i$ and resource $\alpha$), and they all
share a common metabolic matrix $\mathbf{D}$.
The element $D_{\alpha\beta}$
of this matrix represents the fraction of energy in the form of resource $\alpha$
secreted when resource $\beta$ is consumed.
Here we implemented a new operation mode
in which species can secrete different metabolites (and/or
in different abundances) when consuming a same resource. Experimental observations
support the idea of distinct species producing different sets of byproducts when
growing in the same primary resource \mr. We call $D_{i\alpha\beta}$ to the
fraction of energy in the form of resource $\alpha$ secreted \textit{by species
i} when consuming resource $\beta$ ---note that now $D_{i\alpha\beta}$ need not be
equal to $D_{j\alpha\beta}$ if $i \neq j$, unlike in the original Community
Simulator. In the package's underlying Microbial Consumer Resource Model
\cite{Goldford2018,Marsland2019}, this just means that the energy flux
$J^{\mathrm{out}}_{i\beta}$ now takes the form

\begin{equation}
J^{\mathrm{out}}_{i\beta} = \sum_\alpha D_{i\beta\alpha} l_\alpha J^{\mathrm{in}}_{i\alpha}
\label{eq:jout}
\end{equation}
%
The documentation for the Community Simulator contains detailed
descriptions of the model, parameters and package use. For the updated package with
the new functionality, see \nameref{datacode}.

For our simulations,
we first generate a library of 660 species (divided into three specialist
families of 200 members each
and a generalist family of 60 members)
and 30 resources (divided into three classes of 10 members each).
We split this library into two non-overlapping pools of 330 species each.
We randomly sample 50 species from each pool in equal ratios to seed
100 resident and
100 invasive communities respectively.
We then grow and dilute the communities serially,
replenishing the primary
resource after each dilution.
We repeat the process 20 times to ensure generational equilibrium is
achieved \cite{Goldford2018}.
We then perform the \textit{in silico} experiments by using the
generationally stable communities to seed 100 coalesced communities
that we again stabilize as described previously.
Similarly, we identify the dominant (most
abundant) species of every resident and invasive community to carry out pairwise
competition and single invasion simulations.
Most parameters are set to the defaults of the original Community Simulator
package. Table \mr shows those that are given non-default values to ensure
enough variation in the primary communities.

\section*{Data \& code availability}\label{datacode}

Experimental data and code for the analysis, as well as code for the simulations
and the updated Community Simulator package with instructions for the
new features can be found in \url{github.com/jdiazc9/coalescence}.

\section*{Acknowledgements}

The authors wish to thank Joshua Goldford, Pankaj Mehta, Wenping Cui,
Robert Marsland and all members of the Sanchez laboratory
for many helpful discussions.
We also wish to express our gratitude to the Goodman laboratory at Yale
for technical help during the early stages of this project.
The funding for this work partly results from a Scialog Program
sponsored jointly by the Research Corporation for Science Advancement and
the Gordon and Betty Moore Foundation through grants to Yale University by the
Research Corporation and the Simons Foundation.

\clearpage

% references
\clearpage
\bibliographystyle{mystyle}
\bibliography{refs}

\clearpage

\section*{Figures}\label{figs}

\begin{figure}[!h]
\centering
\internallinenumbers
\includegraphics[width=10cm,keepaspectratio]{figs/fig1.pdf}
\caption{\textbf{Overview of the experimental protocol.}
\textbf{a.}~Environmental samples collected from eight different locations were used
to inoculate our communities.
\textbf{b.}~Communities were stabilized in serial batch culture bioreactors
\cite{Goldford2018} in minimal synthetic media with glutamine or citrate as the
only supplied carbon source.
\textbf{c.}~Communities were plated in minimal media agar plates and the most abundant
species (the ``dominants") from each community were isolated. We refer to the set of
sub-dominant species as the ``cohorts".
\textbf{d.}~Rank-frequency distributions of all eight communities stabilized in either
glutamine (red) or citrate (blue), sequenced at a depth of $10^{-4}$ reads.
Three biological replicates per community are shown.
Community compositions are skewed and long-tailed.
\textbf{e.}~Our hypothesis is that ecological co-selection can take place from the top-down,
i.e. the dominant co-selecting the cohort, or from the bottom-up, i.e. the cohort co-selecting
the dominant. Both forms of co-selection can be positive (recruitment) or negative
(antagonism).
\textbf{f.}~Illustration of the protocol of our coalescence experiments. All pairs of
communities were inoculated into fresh minimal media supplemented with the same carbon
source where communities had been previously stabilized. The coalesced (C) and original
resident (R) and invasive (I) communities were then serially diluted and allowed to grow
for seven additional transfers.}
\label{fig1}
\end{figure}

\clearpage

\begin{figure}[!h]
\centering
\internallinenumbers
\includegraphics[width=15cm,keepaspectratio]{figs/fig2.pdf}
\caption{\textbf{Co-selection in microbial community coalescence.}
\textbf{a.}~Coalescence outcomes are quantified by the relative Bray-Curtis similarity
($Q$) between the coalesced and invasive communities (\methodsref{metrics}).
These outcomes are predicted by the pairwise competition between the invasive and
resident dominant
species ($R^2=0.15$ for glutamine and $R^2=0.57$ for citrate). This is consistent with a scenario of
top-down positive co-selection where dominants recruit their cohorts for the final
coalesced community. Two biological replicates per experiment are plotted individually.
\textbf{b.}~We represent the frequency reached by
the invasive dominant species when they invade the resident communities on their own versus
when they are in the company of their cohort.
Three scenarios are possible: green and red
shaded areas represent limit cases of positive (recruitment) or negative (antagonism)
bottom-up co-selection, gray area corresponds to situations where invasive dominant
species can invade with equal success regardless of the presence of their cohorts.
Data shows that positive co-selection is common, whereas antagonistic
co-selection is rare in our experiments.
Two biological replicates per experiment are plotted individually.
\textbf{c-d.} Simulations of community coalescence with a consumer-resource model
are able to capture these trends.}
\label{fig2}
\end{figure}

\clearpage

\section*{Supplementary Figures}

\renewcommand{\thefigure}{S\arabic{figure}}
\setcounter{figure}{0} 

\begin{figure}[!h]
\centering
\internallinenumbers
\includegraphics[width=15cm,keepaspectratio]{figs/figS1.pdf}
\caption{\textbf{Community compositions after seven additional transfers without coalescence.}
Each color of the pie plots corresponds to a different exact sequence variant
(\methodsref{sequencing}).
Replicate 2 of community 1 from glutamine,
as well as replicate 2 of community 8 from citrate (highlighted) were removed based on their
dissimilarity to the other two replicates (details in code for data analysis, see
\nameref{datacode}).
Communities clustered in dashed boxes shared the same dominant species as revealed by
sequencing data.
For communities enclosed in red boxes, sequencing data showed that the species isolated by plating
was not detectable in the community after seven additional transfers (i.e. the dominant was
incorrectly identified) and were therefore excluded from downstream analyses.}
\label{figS1}
\end{figure}

\clearpage

\begin{figure}[!h]
\centering
\internallinenumbers
\includegraphics[width=12.5cm,keepaspectratio]{figs/figS2.pdf}
\caption{\textbf{Alternative metrics of community distance.}
Quantifying coalescence outcomes using different metrics of community similarity
(\methodsref{metrics}) gives similar results to those shown in \figref[a]{fig2}.
Metrics that account for the relative species abundances (Bray-Curtis or Jensen-Shannon similarities)
yield higher correlations than less quantitative metrics that only
account for species presence/absence (Jaccard similarity or the fraction of endemic
invasive species persisting in the coalesced community).
\textbf{a.} Relative Jensen-Shannon similarity 
($R^2=0.15$, $p<0.05$ for glutamine and $R^2=0.53$, $p<5\times10^{-4}$ for citrate)
\textbf{b.} Relative Jaccard similarity
($R^2=0.08$, $p>0.05$ for glutamine and $R^2=0.13$, $p>0.05$ for citrate)
\textbf{c.} Relative survival of invasive endemic species after coalescence
($R^2=0.16$, $p<0.05$ for glutamine and $R^2=0.04$, $p>0.05$ for citrate).}
\label{figS2}
\end{figure}

\clearpage

\begin{figure}[!h]
\centering
\internallinenumbers
\includegraphics[width=10cm,keepaspectratio]{figs/figS3.pdf}
\caption{\textbf{Dominant species have limited effects on coalescence outcomes quantification.}
We repeated the analyses shown in \figref[a]{fig2} and \figref{figS2}, but this time we removed the
dominants from the compositional data prior to quantifying community distances.
The trends observed before are maintained.
\textbf{a.} Relative Bray-Curtis similarity
($R^2=0.20$, $p<0.01$ for glutamine and $R^2=0.34$, $p<0.005$ for citrate)
\textbf{b.} Relative Jensen-Shannon similarity 
($R^2=0.24$, $p<0.005$ for glutamine and $R^2=0.36$, $p<0.005$ for citrate)
\textbf{c.} Relative Jaccard similarity
($R^2=0.09$, $p>0.05$ for glutamine and $R^2=0.11$, $p>0.05$ for citrate)
\textbf{d.} Relative survival of invasive endemic species after coalescence
($R^2=0.18$, $p<0.05$ for glutamine and $R^2=0.08$, $p>0.05$ for citrate).}
\label{figS3}
\end{figure}

\clearpage

\begin{figure}[!h]
\centering
\internallinenumbers
\includegraphics[width=8cm,keepaspectratio]{figs/figS4.pdf}
\caption{\textbf{Pairwise competition of dominants with or without their cohorts.}
In the horizontal axis, we plot the relative frequency of the invasive dominant species in head-to-head
pairwise competition. In the vertical axis, we plot the same relative frequency when competition takes
place in the presence of the cohorts, i.e. during community coalescence.
$R^2=0.04$, $p>0.05$ for glutamine and $R^2=0.83$, $p<10^{-8}$ for citrate.}
\label{figS4}
\end{figure}

\clearpage

\section*{Test section}

This is an example cite \cite{Luo2020,Vass2021}.
This is how you refer to \figref{fig1}.
This is how you refer to \figref[a]{fig1}.
This is how you refer to \tableref{fig1}.
This is how you refer to \figref{figS1}.
This is how you refer to the section \nameref{methods:sim} of the \nameref{methods}.
This is a cleaner way to refer to the \methodsref{sim}.


Check this reference \cite{Castledine2020}

Resource availability modulates biodiversity–invasion relationships by altering competitive interactions \cite{Yang2017}

Resource pulses can alleviate the biodiversity–invasion relationship in soil microbial communities \cite{Mallon2015}

\end{document}




















